pkg/query/storeset.go

```diff
@@ -29,6 +29,16 @@ type StoreSpec interface {
 	Metadata(ctx context.Context, client storepb.StoreClient) (labels []storepb.Label, mint int64, maxt int64, err error)
 }
 
+type StoreStatus struct {
+	Name        string
+	Healthy     bool
+	LastCheck   time.Time
+	LastMessage string
```

Comment:
Removed the newly message and added `Error` instead of `string`.

I like the idea of having a link to the state ! Though itI'll have to select the store via their labels: 
```
thanos_store_node_info{external_labels="{federator_role=\"xxx\",replica=\"AB\"}"} 1
```